src/python/grpcio_tests/tests_aio/unit/close_channel_test.py

```diff
@@ -0,0 +1,170 @@
+
+            def time_remaining(self):
+                raise NotImplementedError
+
+        ongoing_calls = _OngoingCalls()
+        self.assertEqual(ongoing_calls.size(), 0)
+
+        call = FakeCall()
+        ongoing_calls.trace_call(call)
+        self.assertEqual(ongoing_calls.size(), 1)
+        self.assertEqual(ongoing_calls.calls, [call])
+
+        call.callback(call)
+        self.assertEqual(ongoing_calls.size(), 0)
+        self.assertEqual(ongoing_calls.calls, [])
+
+
+class TestCloseChannel(AioTestBase):
+
+    async def setUp(self):
+        self._server_target, self._server = await start_test_server()
+
+    async def tearDown(self):
+        await self._server.stop(None)
+
+    async def test_graceful_close(self):
+        channel = aio.insecure_channel(self._server_target)
+        UnaryCallWithSleep = channel.unary_unary(
+            _UNARY_CALL_METHOD_WITH_SLEEP,
+            request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+            response_deserializer=messages_pb2.SimpleResponse.FromString,
+        )
+
+        call = UnaryCallWithSleep(messages_pb2.SimpleRequest())
+        task = asyncio.ensure_future(call)
+
+        await channel.close(grace=UNARY_CALL_WITH_SLEEP_VALUE * 2)
+
+        self.assertEqual(grpc.StatusCode.OK, await call.code())
+
+    async def test_none_graceful_close(self):
+        channel = aio.insecure_channel(self._server_target)
+        UnaryCallWithSleep = channel.unary_unary(
+            _UNARY_CALL_METHOD_WITH_SLEEP,
+            request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+            response_deserializer=messages_pb2.SimpleResponse.FromString,
+        )
+
+        call = UnaryCallWithSleep(messages_pb2.SimpleRequest())
+        task = asyncio.ensure_future(call)
+
+        await channel.close(None)
+
+        self.assertEqual(grpc.StatusCode.CANCELLED, await call.code())
+
+    async def test_close_unary_unary(self):
+        channel = aio.insecure_channel(self._server_target)
+        stub = test_pb2_grpc.TestServiceStub(channel)
+
+        calls = [stub.UnaryCall(messages_pb2.SimpleRequest()) for _ in range(2)]
+
+        self.assertEqual(channel._ongoing_calls.size(), 2)
+
+        await channel.close()
+
+        for call in calls:
+            self.assertTrue(call.cancelled())
+
+        self.assertEqual(channel._ongoing_calls.size(), 0)
+
+    async def test_close_unary_stream(self):
+        channel = aio.insecure_channel(self._server_target)
+        stub = test_pb2_grpc.TestServiceStub(channel)
+
+        request = messages_pb2.StreamingOutputCallRequest()
+        calls = [stub.StreamingOutputCall(request) for _ in range(2)]
+
+        self.assertEqual(channel._ongoing_calls.size(), 2)
+
+        await channel.close()
+
+        for call in calls:
+            self.assertTrue(call.cancelled())
+
+        self.assertEqual(channel._ongoing_calls.size(), 0)
+
+    async def test_close_stream_stream(self):
```

Comment:
Should we also add a test case for stream_unary?

Reply:
ofc, I didn't spot that I was forgetting of one arity :(